### ISSUE TITLE
XAS scans stall periodically

### ISSUE THREAD

**Issue by helfrij** [acquaman/acquaman#2073]
In testing #2071 on Side, I've noticed that XAS scans will periodically "stall", they stop partway through a scan and don't ever seem to resume again. These are ordinary XAS scans with I0, I1, and I2 detectors enabled. It seems to happen at a relatively high frequency, which is troubling.

This problem may be specific to that branch, though I can't think of what change would have caused this. Further investigation required.

**Comment by regiert** [comment]
It may or may not be related, but on SGM map scans using the commissioning tool will periodically stall.  There is no error message and the scan is still active.  I was able to get the scan to resume by manually triggering the detector (in this case amptek SDD4).

**Comment by helfrij** [comment]
@regiert, I suspect the problem seen on BioXAS Side is related to the Zebra box that triggers our acquisitions. However, our XAS scan code does inherit from AMGenericStepScan (commissioning tool) classes. I will keep in mind that there may be a bug in the more general code as I investigate.

**Comment by helfrij** [comment]
I think this may have something to do with changes to the scaler with the Zebra. There is new code there that handles mode switching that treats switches differently than the AcquisitionMode control.

**Comment by helfrij** [comment]
@dretrex and I investigated this today. The most likely cause is that channel 19 was always enabled, even if the diode/PIPS/Lytle aren't being used. The scaler waits for all enabled channels to update before reporting that it has successfully acquired, but the trigger source only waits for the scan detectors (the extra detector missing here).

**Comment by dretrex** [comment]
Turns out that was a red herring.  But we did find the real culprit!  

The .HIGH field was set to 10 ms to allow as wide a range of possible dwell times.  Unfortunately, the pulse blocks of the zebra are edge triggered and what was happening is that the soft inputs were not high long enough to ensure that the "edge" was registered by the application.  For the time being, we have set the .HIGH field to 100 ms and there have been no hung up scans as of yet, which is promising.  Depending on scientific demands, this may need to be adjusted.  Regardless of the demands, 10 ms is certainly too low.
